### PR TITLE
[FIX] mail: display follower subtypes order by sequence

### PR DESCRIPTION
**Pull request by auon-odoo**
Before this fix, follower subtypes were displayed by order of id,
whereas in DiscussController.read_subscription_data(),
it was ordered by :
\- parent_mode
\- res_mode
\- internal
\- sequence

See : https://github.com/odoo/odoo/blob/fc115ebdf7c3783f9eaabba71eccce23c676d957/addons/mail/controllers/discuss.py#L471-472
In this commit, we just reproduce the same order.

Also, for `parentModel` and `resModel`, we want the record
with their values unset to come first.
We added 'undefined-first',
as this sort order might be needed for other models.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/model/model_field_relation_set.js [hunk 1]
```diff
@@ -80,6 +80,18 @@ export class RelationSet {
                                     }
                                     break;
                                 }
+                                case 'undefined-first': {
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_field_relation_set.js:83]: `undefined-first` => `falsy-first`

**Review by alexkuhn — CHANGES_REQUESTED** [review]
LGTM otherwise

File: addons/mail/static/src/model/model_field_relation_set.js (hunk 1 above)
Comment (auon-odoo) [reply addons/mail/static/src/model/model_field_relation_set.js]: Thanks

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**